Server/src/transport/legacy/unity_connection.py

```
@@ -499,6 +544,8 @@ def read_status_file(target_hash: str | None = None) -> dict | None:
                     sleep_s = self._cap_to_deadline(sleep_s, deadline, floor=0.0)
                     time.sleep(sleep_s)
                     continue
+                if target_unavailable_error is not None:
+                    raise target_unavailable_error
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)